.gitignore

```diff
@@ -1,1071 +1,269 @@
-# Global ignores
-__pycache__/
+# Wildcards
+*.8
+*.a
+*.ao
+*.d
+*.dSYM
+*.db
+*.dll
+*.dylib
+*.exp
+*.lib
+*.lnk
+*.ncb
+*.o
+*.obj
+*.opendb
+*.opendf
+*.opensdf
+*.opt
+*.pc
+*.pdb
+*.plg
 *.pyc
+*.s
+*.sdf
+*.so
+*.suo
+*.tgz
+*.user
+*.vcproj.*.*.user
+*.vcxproj.user
+cygicudata*
+derb_*
+ht_*
+libicu*
+libicutest*
+libsicu*
+libsicutest*
+output-*.cpp
+perf-*.xml
+test-*.xml
+
+# Directories
+.vs/
+.vscode/
+ARM/
+ARM64/
+Debug/
+Generated[!!-~]Files/
+Release/
+__pycache__/
+arm/
+arm64/
+bin/
+debug/
+doc/
+lib/
+out/
+release/
+tmp/
+x64/
+x86/
+
+# Files
+Makefile
+Makefile.local
+build-local.properties
+charperf
+collperf
```

Comment:
`charperf` and `collperf` are specific binary names, so probably best to keep in the "specific file names" section, even though there are two of them each.